### DISCUSSION TITLE
项目维护

### DISCUSSION THREAD

**Discussion by wangzhankun** [HARM4Y/FOKS-TROT#30] [General]
关于项目维护建议回滚到c42bc11b67be37f3fc76dff9fdf677b9fb1e57db然后强制推送。我在c42bc11b67be37f3fc76dff9fdf677b9fb1e57db及其之前的代码更改是符合你在最新的README中的规范要求的。你最新的几次提交如果仅仅是对README进行更改了话，我建议是先保存一份README然后强制回退，之后再修改README。要不然现在全部回滚之后，我再修改代码工程量还挺多的。

**Comment by HARM4Y** [comment]
行啊，我同意，回滚到c42bc11b67be37f3fc76dff9fdf677b9fb1e57db吧
今天项目分类修改的太多了，咱们俩一起维护肯定需要多讨论，需要互相尊重对方的付出

**Comment by HARM4Y** [comment]
ok了，git真难用，我之前维护都是直接删掉整个文件夹，确实不妥
老哥，这个项目基本上就交到你的手上了，还是尽量保留我的写过代码的痕迹吧，好让我以后工作时去吹吹牛逼。
你要是愿意继续写点啥，新功能的添加随你。

**Comment by HARM4Y** [comment]
![image](https://user-images.githubusercontent.com/41336794/171411397-7bd91427-7453-433f-9f38-16df7b362aa1.png)
红线处不用再转换为DOS名了，内核内的文件名都是DOS名，其次PocQuerySymbolicLink->ZwOpenSymbolicLinkObject这个函数只能输入符号链接名，如果输入dos名会导致返回失败0xc000003a。因此整个文件夹过滤函数失效。我已修复。
![image](https://user-images.githubusercontent.com/41336794/171412382-e8a6dc04-0149-4b40-94f2-0e66382dfaba.png)
这里也不能用FltGetFileNameInformation这个函数，请查询msdn。
代码写完以后请测试再上传github？第一个问题会导致整个minifilter失效。
@wangzhankun

**Reply by wangzhankun** [reply]
图中画红线的部分是可以这样调用的。PocAnyPath2DosPath我做了特判。主要是在最上层调用的时候，因为是用户通过CommPort传过来的路径，懒得做路径转换了。我提供PocAnyPath2DosPath的目的也是调用者可以通过该函数将任意格式转为DosPath而不用特殊考虑其路径格式，相当是把`PocSymbolLinkPathToDosPath`和`PocQuerySymbolicLink`做了进一步的封装。

**Reply by wangzhankun** [reply]
第二个注释掉是可以的，那个是我为了方便断点写的特判。

**Comment by HARM4Y** [comment]
```		
else if (wcsstr(src_path, L"\\device\\harddiskvolume") != NULL)
{
	RtlMoveMemory(dest_path, src_path, wcslen(src_path) * sizeof(WCHAR));
}
```	
wcsstr是大小写敏感的，我已修复这里
@wangzhankun

**Comment by HARM4Y** [comment]
docx的加密出现了一个大bug，可能是14号以后我更新出现的问题，所以最近没有更新的打算了，我要花时间解决这个bug，暂时将你移除collaborators，请你见谅。 @wangzhankun

**Reply by wangzhankun** [reply]
没问题，我先把提交放到我的仓库那里。之后你开通权限之后我再合并回来。

**Reply by HARM4Y** [reply]
好的

**Comment by wangzhankun** [comment]
关于头文件中函数声明的注释规范我建议写成如下的格式：
```cpp
/**
 * @Author: wangzhankun
 * @Date: 2022-06-04 18:25:32
 * @LastEditors: wangzhankun
 * @update: 
 * @brief: 向list的末尾添加一个节点。
 * @param {TSDL_ENTRY} *head
 * @param {TSDL_ENTRY} *node
 * @return {PTSDL_ENTRY}
 */
PTSDL_ENTRY TSDL_AddTail(TSDL_ENTRY *head, TSDL_ENTRY *node);

```

**Reply by HARM4Y** [reply]
这么高级的嘛

**Comment by HARM4Y** [comment]
[FOKS-TROT.docx](https://github.com/hkx3upper/FOKS-TROT/files/8849916/FOKS-TROT.docx)
 同学，你愿意完善一下开发文档吗？这个版本的开发文档有不少错误，你可以从头审阅一下，然后加些新的。正好可以把有些不确定的地方批注一下，写完以后在discussion@我一下，我补充解释一下。
@wangzhankun

**Reply by HARM4Y** [reply]
我已更新

**Reply by wangzhankun** [reply]
抱歉，前段时间在考试。我在想这个文档是否可以整理成MARKDOWN的格式放到项目里面?如果可以的话我整理一下。

**Reply by HARM4Y** [reply]
要是太麻烦就不用了，没啥必要

**Reply by wangzhankun** [reply]
同学，Wiki的权限你帮我开一下吧。我提交不上去。
